src/wavm.cpp

```diff
@@ -268,21 +275,28 @@ ExecutionResult WavmEngine::internalExecute(
   ensureCondition(mainFunction, ContractValidationFailure, "\"main\" not found");
 
   // this is how WAVM's try/catch for exceptions
-  Runtime::catchRuntimeExceptions(
-    [&] {
-      try {
-        vector<IR::Value> invokeArgs;
-        Runtime::invokeFunctionChecked(wavm_context, mainFunction, invokeArgs);
-      } catch (EndExecution const&) {
-        // This exception is ignored here because we consider it to be a success.
-        // It is only a clutch for POSIX style exit()
+  // TODO: this is terrible, need to clean up the handling of `interface`
+  try {
+    Runtime::catchRuntimeExceptions(
```

Comment:
Does this actually execute anything?

Reply:
Yeah, it catches wasm runtime errors (but passes through exceptions in host functions).